### PR TITLE
Update the notebook Tutorial-3 according to PR262

### PR DESCRIPTION
**Pull request by arthurPignet**
Update the notebook 3

### CODE REVIEW

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 1]
```diff
@@ -146,33 +146,29 @@
...
     "#### Model generator\n",
-    "This function provides the model use, which will be trained by the scenario object. \n",
+    "This method needs to be implemented, and provides the model use, which will be trained by the scenario object.\n",
     "Note: It is mandatory to have loss and accuracy as metrics for your model.\n",
     "\n",
     "#### Train/validation/test splits\n",
     "\n",
-    "The dataset object must be provided some separated train and test sets (referred to as global train set and global test set).\n",
-    "The global train set is then further split into a global train set and a global validation set, by the function `train_val_split_global`. Please denote that if this function is not provided, the sklearn's train_test_split function will be called by default, and 10% of the training set will be use as validation set. \n",
+    "The Dataset constructor (called via `super()` )  must be provided some separated train and test sets (referred to as global train set and global test set).\n",
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:162]:
```suggestion
    "The `Dataset` constructor (called via `super()`)  must be provided some separated train and test sets (referred to as global train set and global test set).\n",
```

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 2]
```diff
@@ -146,33 +146,29 @@
...
+    "    num_classes,\n",
     ")\n",
     "```\n",
     "#### Data labels\n",
     "The data labels can take whatever shape you need, with only one condition. \n",
     "The labels need to be convertible into string format, and with respect to the condition that if label1 is equal to label2 (\n",
     "reciprocally different from), therefore str(label1) must be equal to str(label2) (reciprocally different from)\n",
     "#### Model generator\n",
-    "This function provides the model use, which will be trained by the scenario object. \n",
+    "This method needs to be implemented, and provides the model use, which will be trained by the scenario object.\n",
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:157]:
```suggestion
    "This method needs to be implemented, and provides the model use, which will be trained by the `scenario` object.\n",
```

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 3]
```diff
@@ -146,33 +146,29 @@
...
     "Note: It is mandatory to have loss and accuracy as metrics for your model.\n",
     "\n",
     "#### Train/validation/test splits\n",
     "\n",
-    "The dataset object must be provided some separated train and test sets (referred to as global train set and global test set).\n",
-    "The global train set is then further split into a global train set and a global validation set, by the function `train_val_split_global`. Please denote that if this function is not provided, the sklearn's train_test_split function will be called by default, and 10% of the training set will be use as validation set. \n",
+    "The Dataset constructor (called via `super()` )  must be provided some separated train and test sets (referred to as global train set and global test set).\n",
+    "The global train set is then further split into a global train set and a global validation set, by the function `train_val_split_global`. Please denote that if this function is not overwritten, the sklearn's train_test_split function will be called by default, and 10% of the training set will be use as validation set.\n",
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:163]:
```suggestion
    "The global train set is then further split into a global train set and a global validation set, by the function `train_val_split_global`. Please denote that if this function is not overwritten, the sklearn's `train_test_split` function will be called by default, and 10% of the training set will be use as validation set.\n",
```

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 4]
```diff
@@ -214,7 +210,7 @@
     "id": "Qf36Veh3fyLU"
    },
    "source": [
-    "#### Define preprocessing functions"
+    "#### Define our class"
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:213]: Dataset class?

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 5]
```diff
@@ -288,141 +356,11 @@
...
-    "\n",
-    "input_shape = max_tokens\n",
-    "num_classes = len(np.unique(y))\n",
+    "#### Create dataset\n",
     "\n",
-    "\n",
-    "print('length of the dictionary ',len(word_index))\n",
-    "print('max token ', max_tokens) \n",
-    "print('num classes', num_classes)"
+    "And we can eventually generate our object !"
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:361]:
```suggestion
    "And we can eventually generate our object!"
```

**Review by natct10 — COMMENTED** [review]
Looks cool, but notebook diffs are not easy to read...

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 6]
```diff
@@ -131,9 +131,9 @@
     "## 3 - Generation, and preparation of the dataset\n",
     " \n",
     "The scenario object needs a dataset object to run. In the previous tutorials, we indicate which one to generate automatically by passing a name of a pre-implemented dataset to the scenario constructor. \n",
-    "Here, we will create this dataset object and pass it to the scenario constructor. \n",
+    "Here, we will create this dataset object and pass it to the scenario constructor. To do so, we are going to create a new class, which inherit from the mplc.Dataset abstract class.\n",
     "\n",
-    "The dataset needs few arguments. \n",
+    "A sub-class of Dataset needs few attribute and method. First, the constructor of the Dataset object needs few arguments..\n",
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:136]:
```suggestion
    "A sub-class of Dataset needs few attribute and method. First, the constructor of the Dataset object needs few arguments.\n",
```

File: notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb [hunk 7]
```diff
@@ -146,33 +146,29 @@
...
     "#### Model generator\n",
-    "This function provides the model use, which will be trained by the scenario object. \n",
+    "This method needs to be implemented, and provides the model use, which will be trained by the `Scenario` object.\n",
     "Note: It is mandatory to have loss and accuracy as metrics for your model.\n",
     "\n",
     "#### Train/validation/test splits\n",
     "\n",
-    "The dataset object must be provided some separated train and test sets (referred to as global train set and global test set).\n",
-    "The global train set is then further split into a global train set and a global validation set, by the function `train_val_split_global`. Please denote that if this function is not provided, the sklearn's train_test_split function will be called by default, and 10% of the training set will be use as validation set. \n",
+    "The `Dataset` constructor (called via `super()` )  must be provided some separated train and test sets (referred to as global train set and global test set).\n",
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-3_Use_homemade_dataset.ipynb:162]:
```suggestion
    "The `Dataset` constructor (called via `super()`)  must be provided some separated train and test sets (referred to as global train set and global test set).\n",
```

**Review by natct10 — APPROVED** [review]
Added 2 suggestions, but it's okay for me, thank you @arthurPignet!

File: notebooks/tutorials/Tutorial-1_Run_your_first_scenario.ipynb [hunk 8]
```diff
@@ -14,8 +14,7 @@
     "In order to run this example, you'll need to:\n",
     "\n",
     "* use python 3.7 +\n",
-    "* install requirements from the requirements.txt file\n",
-    "* install this package https://test.pypi.org/project/mplc/\n"
+    "* install this package https://pypi.org/project/mplc/\n"
```
Comment (natct10) [inline notebooks/tutorials/Tutorial-1_Run_your_first_scenario.ipynb:17]: Bien vu

**Review by bowni — APPROVED** [review]
